allennlp/training/metrics/conll_coref_scores.py

```diff
@@ -150,29 +153,21 @@ def update(self, predicted, gold, mention_to_predicted, mention_to_gold):
         self.recall_denominator += r_den
 
     def get_f1(self):
-        precision = (
-            0
-            if self.precision_denominator == 0
-            else self.precision_numerator / float(self.precision_denominator)
-        )
-        recall = (
-            0
-            if self.recall_denominator == 0
-            else self.recall_numerator / float(self.recall_denominator)
-        )
+        precision = self.get_precision()
+        recall = self.get_recall()
         return 0 if precision + recall == 0 else 2 * precision * recall / (precision + recall)
 
     def get_recall(self):
-        if self.recall_numerator == 0:
+        if self.recall_denominator == 0:
             return 0
         else:
-            return self.recall_numerator / float(self.recall_denominator)
+            return self.recall_numerator / self.recall_denominator
```

Comment:
Not necessary anymore in Python 3 these float castings. I tried checking if they were tensors or not before removing them so as not to remove them if they were.